models/migrations/v78.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-xorm/xorm"
+)
+
+func renameRepoIsBareToIsEmpty(x *xorm.Engine) error {
+	type Repository struct {
+		ID      int64 `xorm:"pk autoincr"`
+		IsBare  bool
+		IsEmpty bool `xorm:"INDEX"`
+	}
+
+	if err := x.Sync(new(Repository)); err != nil {
```

Comment:
Why not just rename column instead of adding/removing?
```
ALTER TABLE "repository" RENAME COLUMN "is_bare" TO "is_empty";
```

for MSSQL you have to use:
```
EXEC sp_rename 'repository.is_bare', 'is_empty', 'COLUMN'
```

Reply:
is the line 

```go
strings.Contains(err.Error(), "no such column") 
```
still correct?

Reply:
hard to tell, need to verify that. Also x.Sync is not needed any more otherwise it will add column and rename will fail